src/Compilers/CSharp/Portable/Symbols/Metadata/PE/SymbolFactory.cs

```diff
@@ -57,6 +57,11 @@ internal override bool IsVolatileModifierType(PEModuleSymbol moduleSymbol, TypeS
             return type.SpecialType == SpecialType.System_Runtime_CompilerServices_IsVolatile;
         }
 
+        internal override bool IsConstModifierType(PEModuleSymbol moduleSymbol, TypeSymbol type)
+        {
+            return type.ToDisplayString() == WellKnownTypes.GetMetadataName(WellKnownType.System_Runtime_CompilerServices_IsConst);
+        }
```

Comment:
> Comparing symbols by full name is probably not be the best way to do it. Looking up a better alternative.

Given that this is not a SpecialType, I think name comparison is the only right way to do that. If performance is a concern we can compare type name and namespace names separately, then calling ToDisplayString won't be needed.